process/factory/interceptorscontainer/sovereignShardInterceptorsContainerFactory.go

```diff
@@ -41,20 +42,107 @@ func NewSovereignShardInterceptorsContainerFactory(
-	_, _, err := sicf.shardInterceptorsContainerFactory.Create()
+	err := sicf.generateTxInterceptors()
 	if err != nil {
 		return nil, nil, err
 	}
 
-	err = sicf.generateSovereignHeaderInterceptors()
+	err = sicf.generateUnsignedTxsInterceptors()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	err = sicf.generateHeaderInterceptors(core.SovereignChainShardId)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	err = sicf.generateMiniBlocksInterceptors()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	err = sicf.generateValidatorAndAccountTrieNodesInterceptors(core.SovereignChainShardId)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	err = sicf.generatePeerAuthenticationInterceptor()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	err = sicf.generateHeartbeatInterceptor()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	err = sicf.generatePeerShardInterceptor()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	err = sicf.generateValidatorInfoInterceptor()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	err = sicf.generateSovereignExtendedHeaderInterceptors()
 	if err != nil {
 		return nil, nil, err
 	}
 
 	return sicf.mainContainer, sicf.fullArchiveContainer, nil
 }
 
-func (sicf *sovereignShardInterceptorsContainerFactory) generateSovereignHeaderInterceptors() error {
+func (sicf *sovereignShardInterceptorsContainerFactory) generateTxInterceptors() error {
+	keys := make([]string, 0, 1)
+	interceptorSlice := make([]process.Interceptor, 0, 1)
+
+	identifierTx := factory.TransactionTopic + sicf.shardCoordinator.CommunicationIdentifier(core.SovereignChainShardId)
+	interceptor, err := sicf.createOneTxInterceptor(identifierTx)
+	if err != nil {
+		return err
+	}
+
+	keys = append(keys, identifierTx)
+	interceptorSlice = append(interceptorSlice, interceptor)
+	return sicf.addInterceptorsToContainers(keys, interceptorSlice)
+}
+
+func (sicf *sovereignShardInterceptorsContainerFactory) generateUnsignedTxsInterceptors() error {
+	keys := make([]string, 0, 1)
+	interceptorsSlice := make([]process.Interceptor, 0, 1)
+
+	identifierScr := factory.UnsignedTransactionTopic + sicf.shardCoordinator.CommunicationIdentifier(core.SovereignChainShardId)
+	interceptor, err := sicf.createOneUnsignedTxInterceptor(identifierScr)
+	if err != nil {
+		return err
+	}
+
+	keys = append(keys, identifierScr)
+	interceptorsSlice = append(interceptorsSlice, interceptor)
+
+	return sicf.addInterceptorsToContainers(keys, interceptorsSlice)
+}
+
+func (sicf *sovereignShardInterceptorsContainerFactory) generateMiniBlocksInterceptors() error {
+	keys := make([]string, 1)
+	interceptorsSlice := make([]process.Interceptor, 1)
+
+	identifierMiniBlocks := factory.MiniBlocksTopic + sicf.shardCoordinator.CommunicationIdentifier(core.SovereignChainShardId)
+	interceptor, err := sicf.createOneMiniBlocksInterceptor(identifierMiniBlocks)
+	if err != nil {
+		return err
+	}
+
+	keys[0] = identifierMiniBlocks
```

Comment:
Use the same style of code as previous functions.

Reply:
Will clean it up in future PRs